files/aws/lib/systemd/system/systemd-networkd.service.d/override.conf

```diff
@@ -0,0 +1,3 @@
+[Service]
+ExecStartPre=
+ExecStartPre=+/opt/delphix/server/bin/update_netplan_to_mac_address
```

Comment:
yea, I think it might be better to create this file via the app-gate, due to the dependency issue.. presumably, the `postinst` logic could go in the app-gate's script too..

Reply:
The dependency has been broken by moving the script update_netplan_to_mac_address to delphix-platform repository (it was earlier insider dlpx-app-gate repo). This should allow the override.conf file and the update_netplan_to_mac_address script to co-exist without any dependency on delphix-virtualization.

The default override.conf file (common across all the platforms) has been renamed to 10-override.conf. The new override.conf file with the PreExecStart directive has been renamed to 20-override.conf file. With these two changes, we ensure that the default and common 10-override.conf file always executes first, and the 20-override.conf file will only execute only in AWS environment.

Reply:
Do these changes address all the concerns ?

Reply:
Yes, but if we're renaming these, I'd prefer to give them meaningful names (e.g. you could keep the default override file name the same, and name the new one aws-override.conf or update-interface-names.conf, etc.). I don't think that the order matters in this case (i.e. I'm not sure why we'd use a "10" or "20" prefix for each).

Reply:
To test the current changes, ,I added a dummy override.conf file to the systemd-networkd service and the systemctl cat command lists it at the end --

delphix@ip-10-110-237-180:~$ sudo systemctl cat systemd-networkd | grep "^# /"
# /lib/systemd/system/systemd-networkd.service
# /usr/lib/systemd/system/systemd-networkd.service.d/10-override.conf
# /usr/lib/systemd/system/systemd-networkd.service.d/20-override.conf
# /usr/lib/systemd/system/systemd-networkd.service.d/override.conf

I believe the override files will get processed in the order above. We would ideally want the default override.conf file to be processed before any platform specific override files. Prefixing the filenames with digits makes it easy to achieve so.

Reply:
I thought that we had done an experiment a few weeks ago to determine that the order of the sections included in these files doesn't matter...(?)

Reply:
FWIW, I have no problem with giving them an explicit ordering, I'm just making sure I understand the reasoning. My point was also about the naming; I think you can name them anything as long as they have a `.conf` suffix.

Reply:
The aws specific override configuration file has been renamed to aws-override.conf file.
The reasoning behind prefixing override configuration files was discussed with Seb and we agreed to continue the names as suggested in the PR. The naming scheme will ensure that the default override.conf file will be processed before any platform specific override files.